src/main/java/seedu/address/model/person/Schedule.java

```diff
@@ -0,0 +1,133 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+/**
+ * Represents a Person's schedule in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}, {@link #isValidDate(String)},
+ * {@link #isValidTime(String)}
+ */
+public class Schedule {
+
+    public static final String SCHEDULE_NAME_CONSTRAINTS =
+            "Schedule name field must be strictly alphanumeric.";
+    public static final String DATE_CONSTRAINTS =
+            "Date field must be given in the format yyyy-MM-dd.";
+    public static final String TIME_CONSTRAINTS =
+            "Time field must be given in the format HH:mm.";
+
+    public static final String NAME_VALIDATION_REGEX = "\\p{Alnum}+";
+
+    public final LocalDate date;
+    public final LocalTime time;
+    public final String scheduleName;
+    public final String dateString;
+    public final String timeString;
+
+    /**
+     * Constructs a {@code Schedule}
+     *
+     * @param scheduleName Name of schedule, if any
+     * @param date Date of Schedule, given in yyyy-MM-dd format
+     * @param time Time of Schedule, given in HH:mm format, if any
+     */
+    public Schedule(String scheduleName, String date, String time) {
+        requireAllNonNull(scheduleName, date, time);
+
+        checkArgument(isValidName(scheduleName), SCHEDULE_NAME_CONSTRAINTS);
+        checkArgument(isValidDate(date), DATE_CONSTRAINTS);
+        checkArgument(isValidTime(time), TIME_CONSTRAINTS);
+
+        this.date = (date.isEmpty()) ? LocalDate.MIN : LocalDate.parse(date);
+        this.time = (time.isEmpty()) ? LocalTime.MIDNIGHT : LocalTime.parse(time);
+
+        this.scheduleName = (scheduleName.isEmpty()) ? "schedule" : scheduleName;
```

Comment:
Does line 48-51 work as  checkArgument(isValidName(...)) eg. will throw an error for empty strings?

Reply:
isValidName accepts empty strings, since empty strings will simply name the schedule as `schedule`.
It should return false if the schedule provided has non-alphanumeric characters, but I will run some tests to double check